main.js

```diff
@@ -4666,8 +4683,31 @@ function init(compactGroupId) {
             states.subscribe('system.adapter.*.alive');
 
             // set current Loglevel and subscribe for changes
-            states.setState(hostObjectPrefix + '.logLevel', {val: config.log.level, ack: true, from: hostObjectPrefix});
-            states.subscribe(hostObjectPrefix + '.logLevel');
+            states.setState(`${hostObjectPrefix}.logLevel`, {val: config.log.level, ack: true, from: hostObjectPrefix});
+            states.subscribe(`${hostObjectPrefix}.logLevel`);
+
+            try {
+                const nodeVersion = process.version.replace(/^v/, '');
+                const prevNodeVersionState = await states.getStateAsync(`${hostObjectPrefix}.nodeVersion`);
+
+                if (prevNodeVersionState && prevNodeVersionState.val !== nodeVersion) {
+                    // detected a change in the nodejs version
+                    logger.info(`${hostLogPrefix} Node.js version has changed from ${prevNodeVersionState.val} to ${nodeVersion}`);
+                    if (os.platform() === 'linux') {
+                        // ensure capabilities are set
+                        await cpPromise.exec(`sudo setcap cap_net_admin,cap_net_bind_service,cap_net_raw+eip ${process.execPath}`);
```

Comment:
To have it more generic should we make the "setCap" in general into tools so that also other adapters can use that "method" for other executables?